x/staking/client/rest/grpc_query_test.go

```diff
@@ -0,0 +1,101 @@
+package rest_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/cosmos-sdk/testutil"
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	grpctypes "github.com/cosmos/cosmos-sdk/types/grpc"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	cfg     network.Config
+	network *network.Network
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	cfg := network.DefaultConfig()
+	cfg.NumValidators = 1
+
+	s.cfg = cfg
+	s.network = network.New(s.T(), cfg)
+
+	_, err := s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TearDownSuite() {
+	s.T().Log("tearing down integration test suite")
+	s.network.Cleanup()
+}
+
+func (s *IntegrationTestSuite) TestQueryValidatorsGRPCHandler() {
+	val := s.network.Validators[0]
+	baseURL := val.APIAddress
+
+	testCases := []struct {
+		name    string
+		url     string
+		headers map[string]string
+		error   bool
+	}{
+		{
+			"test query validators gRPC route with invalid status",
+			fmt.Sprintf("%s/cosmos/staking/v1beta1/validators?status=active", baseURL),
+			map[string]string{
+				grpctypes.GRPCBlockHeightHeader: "1",
+			},
+			true,
+		},
+		{
+			"test query validators gRPC route without status query param",
```

Comment:
Hmm, if the status query param is required, would it make more sense to have the route as `/cosmos/staking/v1beta1/validators/{status}`?

Reply:
Yes I also agree that requiring query parameters seems strange. Is this a pattern we follow elsewhere? Would be nice to not have it at all as a pattern.

Reply:
That will conflict with other route: `/cosmos/staking/v1beta1/validators/{validator_id}`. I also spent some time to update this in annotations PR but ended up here. Old REST querier is also using it in the same way. May be we can add a default status for this. By default we can fetch all the bonded validators, wdyt?

Reply:
Having a default status sounds better than erroring out no a REST call without query parameters IMO.